src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PathSeqFilterSpark.java

```diff
@@ -0,0 +1,192 @@
+import org.seqdoop.hadoop_bam.BAMInputFormat;
+import scala.Tuple2;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(summary = "Read preprocessing and host filtering on reads from a BAM file",
+        oneLineSummary = "PathSeqFilter on Spark",
+        programGroup = SparkProgramGroup.class)
+public final class PathSeqFilterSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+    private SAMFileHeader header;
+
+    private static final String maxLengthArgName = "pathseq_maxReadLength";
+    @Argument(fullName = maxLengthArgName,
+            shortName = maxLengthArgName,
+            doc="Keep only reads with length at most equal to the specified value",
+            optional=true)
+    public int MAX_READ_LENGTH = 1000;
+
+    private static final String minLengthArg = "pathseq_minReadLength";
+    @Argument(fullName = minLengthArg,
+            shortName = minLengthArg,
+            doc="Keep only reads with length at least equal to the specified value",
+            optional=true)
+    public int MIN_READ_LENGTH = 31;
+
+    @Argument(fullName="pathseq_ambigFilterFrac", shortName="pathseq_ambigFilterFrac", optional=true)
+    public float FRAC_N_THRESHOLD = 0.03f;
+
+    @Argument(fullName="pathseq_baseQualThresh", shortName="pathseq_baseQualThresh", optional=true)
+    public int QUAL_PHRED_THRESH = 15;
+
+    @Argument(fullName="pathseq_baseQualFrac", shortName="pathseq_mbaseQualFrac", optional=true)
+    public float QUAL_MAX_FRAC = 0.05f;
+
+    @Argument(fullName="pathseq_kmerLibPath", shortName="pathseq_kmerLibPath", optional=false)
+    public String KMER_LIB_PATH;
+
+    @Argument(fullName="pathseq_kmerSize", shortName="pathseq_kmerSize", optional=true)
+    public int KMER_SIZE = 31;
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = false)
+    public String output;
+
+    @ArgumentCollection
+    protected OpticalDuplicatesArgumentCollection opticalDuplicatesArgumentCollection = new OpticalDuplicatesArgumentCollection();
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void runTool(final JavaSparkContext ctx) {
+
+        header = getHeaderForReads();
+        ctx.hadoopConfiguration().setBoolean(BAMInputFormat.KEEP_PAIRED_READS_TOGETHER_PROPERTY, true);
+
+        final JavaRDD<GATKRead> reads = getReads();
+
+        //Filter secondary/supplementary reads and reads that fail the vendor quality check
+        JavaRDD<GATKRead> primaryReads = reads.filter(read -> !(read.isSecondaryAlignment() || read.failsVendorQualityCheck() || read.isSupplementaryAlignment()));
```

Comment:
@mwalker174 If this tool had an override of getDefaultFilter, it could just add the corresponding built-in filters to the set returned there. This would make them subject to being enabled, disabled, and configured via their parameters by the user from the command line, and they would be automatically applied by the engine (you could then remove this line completely - the call to getReads() right above returns the RDD with such filter handling already applied). In theory we could do that for all of the filters below, but it looks like there are some like these that are applied before transformations, then some map/transforms, then some post filtering. We've talked about embracing that as pattern in the engine - this tool may be a good use-case for doing so.